src/compiler/checker.rs

```diff
@@ -2838,6 +3230,80 @@ impl<'type_checker> CheckTypeRelatedTo<'type_checker> {
         result
     }
 
+    fn has_excess_properties(
+        &self,
+        source: Rc<Type /*FreshObjectLiteralType*/>,
+        target: Rc<Type>,
+        report_errors: bool,
+    ) -> bool {
+        if !self
+            .type_checker
+            .is_excess_property_check_target(target.clone())
+            || false
+        {
+            return false;
+        }
+        let is_comparing_jsx_attributes = false;
```

Comment:
I guess I'd never really thought about how TS is actually aware of JSX, rather than it being pre-processed into normal TS before it hits the compiler. (That actually seems kind of odd to me.)

Reply:
Right, I didn’t even think about that aspect. So it must basically do the expansion to a function, type-check it, and then apply errors back to the JSX syntax?